### PR TITLE
Anki 24.10rc2 support

### PR DESCRIPTION
**Pull request by voczi**
See title.

Related PRs:
* https://github.com/ankidroid/Anki-Android/pull/17170

For anyone who wants to test the build: https://github.com/voczi/Anki-Android/releases/tag/2.19rc1

### CODE REVIEW

**Review by BrayanDSO — APPROVED** [review]
LGTM. Thanks!

**Comment by mikehardy** [comment]
0.1.41-anki24.10beta2 will be available in a little while (once publish succeeds and maven repo resynchs) if you want to rebase that commit that moves the dep to point to the new one

**Comment by voczi** [comment]
Backend method "compareAnswer" changed: https://github.com/ankitects/anki/pull/3422

**Comment by mikehardy** [comment]
commit for the beta labeled "test integration" for a reason :-)

I'm only typing this part right here because you are very responsive + quick on these normally: I just want to say that I consider this sort of thing "important, but not urgent". That is I love that we are *trying* to see how it will work, but I don't see CI fails here and things breaking to be any sort of emergency at all. Quite the opposite - totally expected, and no need to rush in and fix immediately or anything. Just as it's interesting + you have time. Cheers

**Comment by mikehardy** [comment]
Switched this to beta3 only so I could run my refreshed APK size checker on it, I was curious nothing more.
No action needed or implied as expected or anything

**Comment by twwn** [comment]
I hope it's okay to ask: Is this roughly a 2.19 target, a 2.20 one, a "when Anki 24.10 releases" one, …?

**Comment by mikehardy** [comment]
Hi there @twwn ! This is not the place to ask release timeline prediction questions, which are impossible to predict anyway
In general we always release the current stable version of anki in our current stable version as a goal, and that's about as predictive as we can be - you may infer from that when it will happen

File: AnkiDroid/src/main/java/com/ichi2/anki/browser/CardBrowserColumn.kt [hunk 1]
```diff
@@ -108,15 +108,17 @@ enum class CardBrowserColumn(val ankiColumnKey: String) {
...
+    ORIGINAL_POSITION("originalPosition");
 
     companion object {
 
         val COLUMN1_KEYS = arrayOf(QUESTION, SFLD)
 
         // list of available keys in mCards corresponding to the column names in R.array.browser_column2_headings.
         // Note: the last 6 are currently hidden
-        val COLUMN2_KEYS = arrayOf(ANSWER, CARD, DECK, NOTE_TYPE, QUESTION, TAGS, LAPSES, REVIEWS, INTERVAL, EASE, DUE, CHANGED, CREATED, EDITED)
+        val COLUMN2_KEYS = arrayOf(ANSWER, CARD, DECK, NOTE_TYPE, ORIGINAL_POSITION, QUESTION, TAGS, LAPSES, REVIEWS, INTERVAL, EASE, DUE, CHANGED, CREATED, EDITED)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/browser/CardBrowserColumn.kt:121]: Given the unit test changes, can you confirm that:

* saved columns are not affected by a bump to 24.10
* `R.array.browser_column2_headings` is no longer in-use, or if it is the columns still align correctly
  * I don't recall whether my changes to handle this were merged or not

Comment (voczi) [reply AnkiDroid/src/main/java/com/ichi2/anki/browser/CardBrowserColumn.kt:121]: This felt like something I have overlooked, yeah. Can't we make the fact that the column order has to be retained in the COLUMN2_KEYS assignment a bit more, idk, explicit? Maybe just a comment?

**Comment by voczi** [comment]
Still some other details that I need to iron out. I'll get back to it at the end of this week.

**Comment by voczi** [comment]
Should be better now. Still a bit unsure about the column tests (is there a point of even testing oldData==newData for that column?)

**Review by david-allison — APPROVED** [review]
Looks great, cheers!

Alpha testers will quickly pick up any further issues

@mikehardy have we cut for 2.19/2.20 yet? This would be a great opportunity

**Comment by mikehardy** [comment]
We have not *quite* yet but are right on the point of the branch making sense
2.19.1 just went to beta (finally) yesterday overnight, letting it sit through the day and doesn't look like anything terrible this time (as opposed to my scroll hack oops last time)
If it starts going public (it should?) then I branch and we merge this and dependency updates and we move right along

**Comment by user1823** [comment]
> If it starts going public (it should?) then I branch and we merge this and dependency updates and we move right along

Having a new branch for 2.20 and releasing an alpha soon would be great. Even though people can technically use 24.10 and 24.06 simultaneously, FSRS would work sub-optimally for such users in comparison to those who are using 24.06 on both devices. 

So, for Anki beta-testers and early adopters of the next stable Anki release, it makes sense to have an AnkiDroid alpha at least.

**Comment by mikehardy** [comment]
The branch and alpha will happen shortly, yes
I'm certain many people will be thoroughly excited by the optimal optimalness of the optimal cross-workings of the betas but I won't lose perspective that there's 3.5million devices with Anki on it and I bet 3,499,000 don't even know it's a thing much less care 😆  (assuming we just take care of it for them, which we'll continue doing)
The 1,000 that do are of course all the alpha and beta testers, we'll get them the fresh newness soon enough

**Review by mikehardy — APPROVED** [review]
LGTM to me as well - I need to squash this as i don't believe any of it will work without the ankiBackend version bump, but that's not a problem for me

Have not branched yet but the idea is release-2.19 branch will peel off at the version bump commit for 2.19.1, and that will be performed retroactively, as needed when I do the first cherry-pick for 2.19.2

Thanks for working through this and getting the 24.10 changes all handled, much appreciated

**Pull request merged**